zone/zonedb.cpp

```diff
@@ -1459,45 +1443,32 @@ bool ZoneDatabase::SaveCharacterData(
 	return true;
 }
 
-bool ZoneDatabase::SaveCharacterCurrency(uint32 character_id, PlayerProfile_Struct* pp){
-	if (pp->copper < 0) { pp->copper = 0; }
-	if (pp->silver < 0) { pp->silver = 0; }
-	if (pp->gold < 0) { pp->gold = 0; }
-	if (pp->platinum < 0) { pp->platinum = 0; }
-	if (pp->copper_bank < 0) { pp->copper_bank = 0; }
-	if (pp->silver_bank < 0) { pp->silver_bank = 0; }
-	if (pp->gold_bank < 0) { pp->gold_bank = 0; }
-	if (pp->platinum_bank < 0) { pp->platinum_bank = 0; }
-	if (pp->platinum_cursor < 0) { pp->platinum_cursor = 0; }
-	if (pp->gold_cursor < 0) { pp->gold_cursor = 0; }
-	if (pp->silver_cursor < 0) { pp->silver_cursor = 0; }
-	if (pp->copper_cursor < 0) { pp->copper_cursor = 0; }
```

Comment:
We need to keep this, this was here because we had weird overflow bugs we didn't know where they were coming from and players were getting tons of money. Either keep as is or replace with clamps

Reply:
Never mind you made a function out of it